### PR TITLE
[Meta]: Provide diff of `variables.css`

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Fixes https://github.com/brave/leo/issues/128

This works via a couple of steps:
1. ~~Upload the `variables.css` file as part of the storybook. This means we can fetch the current tokens from https://leo.bravesoftware.com/css/variables.css~~ Done as part of #315 
2. Run a diff of that `variables.css` file and the current one
3. Upload it, as part of a details/summary, so we can see the diff on the PR.

### CODE REVIEW

**Comment by fallaciousreasoning** [comment]
@petemill mind taking a look at this?

**Comment by fallaciousreasoning** [comment]
@kim0 @petemill PTAL

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -55,6 +55,14 @@ jobs:
         run: |
           npm run build
 
+      - name: Diff Tokens
```
Comment (kim0) [inline .github/workflows/build.yml:58]: Not sure why we're having to manually download that file and diff against it. Doesn't git already know what changed. For example, wouldn't something like `git diff ./tokens/css/variables.css` get the diff assuming that file changed ?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -55,6 +55,14 @@ jobs:
...
+        id: diff-tokens
+        run: |
+          curl https://leo.bravesoftware.com/css/variables.css > ./tokens/css/variables.old.css
```
Comment (kim0) [inline .github/workflows/build.yml:61]:
```suggestion
          curl -fsSL -o ./tokens/css/variables.old.css https://leo.bravesoftware.com/css/variables.css
```

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -55,6 +55,14 @@ jobs:
...
+          # Note: Diff has exit code 1 for differences, and there are always
+          # some differences because the file contains a timestamp.
+          diff -u ./tokens/css/variables.old.css ./tokens/css/variables.css > ./tokens/css/variables.diff || echo "Diff Completed"
```
Comment (kim0) [inline .github/workflows/build.yml:64]:
```suggestion
          diff -u ./tokens/css/variables.old.css ./tokens/css/variables.css > ./tokens/css/variables.diff || :
```

Comment (kim0) [reply .github/workflows/build.yml:64]: If you just want to silence `diff`, I think `|| :` makes the intent clearer

File: .github/workflows/build.yml (hunk 1 above)
Comment (fallaciousreasoning) [reply .github/workflows/build.yml:58]: This file is generated as part of the build process. Basically, we want to make it visible on the PR how the generated file is changing.

File: .github/workflows/build.yml (hunk 3 above)
Comment (fallaciousreasoning) [reply .github/workflows/build.yml:64]: That is exactly what I wanted to do but my bash-fu is weak :laughing: Thanks @kim0

**Review by kim0 — APPROVED** [review]

**Review by AlanBreck — APPROVED** [review]
Clever!

**Pull request merged**